qiskit/transpiler/passes/optimization/optimize_1q_decomposition.py

```diff
@@ -102,7 +84,13 @@ def run(self, dag):
                 new_circs.append(decomposer._decompose(operator))
             if new_circs:
                 new_circ = min(new_circs, key=len)
-                if len(run) > len(new_circ) or (single_u3 and new_circ.data[0][0].name != "u3"):
+                if all(g.name in self.basis for g in run) and len(run) < len(new_circ):
+                    warnings.warn(f"Resynthesized {run} and got {new_circ}, "
+                                  f"but the original was native and the new "
+                                  f"value is longer.  This indicates an "
+                                  f"efficiency bug in synthesis.  Please "
+                                  f"report it!")
```

Comment:
Maybe add the link to the bug tracker:

```suggestion
                                  f"report it by opening an issue here:"
                                  f"https://github.com/Qiskit/qiskit-terra/issues/new/choose")
```

Reply:
Also maybe we want to set a stacklevel so the warning doesn't show the warn line, maybe `stacklevel=2` so it shows the `optimize_1q_decomposition.run()` call).